src/inmanta/agent/forking_executor.py

```diff
@@ -163,7 +169,40 @@ def connection_lost(self, exc: Exception | None) -> None:
 
 
 class ExecutorClient(FinalizingIPCClient[ExecutorContext], LogReceiver):
-    pass
+    def __init__(self, name: str):
+        super().__init__(name)
+
+        # Keeps track of when this client was active last
+        self.last_used_at: datetime.datetime = datetime.datetime.now().astimezone()
+
+    def get_idle_time(self) -> timedelta:
+        return datetime.datetime.now().astimezone() - self.last_used_at
+
+    @typing.overload
+    def call(
+        self, method: IPCMethod[ExecutorContext, ReturnType], has_reply: typing.Literal[True] = True
+    ) -> Future[ReturnType]: ...
+
+    @typing.overload
+    def call(self, method: IPCMethod[ExecutorContext, ReturnType], has_reply: typing.Literal[False]) -> None: ...
+
+    @typing.overload
+    def call(self, method: IPCMethod[ExecutorContext, ReturnType], has_reply: bool = True) -> Future[ReturnType] | None: ...
```

Comment:
Isn't this one overload too many?

Reply:
Looks like you added it to the parent as well, so it doesn't seem like a mistake. Why do we need it?

Reply:
Without this overload, mypy will complain that some overloads are missing because it doesn't realize that all cases are covered (see discussion [here](https://github.com/inmanta/inmanta-core/pull/7765#discussion_r1658262146))